_posts/2020-09-19-github-actions-with-faasd.md

```diff
@@ -0,0 +1,256 @@
+---
+title: "Github Actions for OpenFaas functions with faasd"
+description: "Bring CI-CD with Github Actions to your OpenFaaS functions running on a faasd appliance"
+date: 2020-09-19
+image: /images/2020-09-19-github-actions-faasd/lego.jpg
+categories:
+  - faasd
+  - containerd
+  - cicd
+  - github
+author_staff_member: utsav
+dark_background: true
+---
+
+Learn how to bring CI-CD to your OpenFaaS functions running on a faasd appliance with Github Actions
+
+## A few words about faasd
+
+OpenFaaS was created to have the freedom to run serverless functions anywhere, be it on a Raspberry Pi, in a data center, or the cloud. This started with faas-swarm to run on Docker Swarm and faas-netes to run on Kubernetes. faas-netes is the current recommendation for running OpenFaaS in production.
+Though, it's not necessary that you might want or even infact need Kubernetes, especially when considering the complexities involved with managing a Kubernetes cluster, and that's how faasd was born.
+faasd is just like OpenFaaS with the same ecosystem but it spares you from the complexity involved that comes with mangaging a distributed system.
+
+In this tutorial, we'll see how we can get OpenFaaS functions up and running on a faasd appliance with Github Actions.
+
+
+## Prerequisites
+
+* A VM exposed to the Internet
+
+For this tutorial we'll be using a [DigitalOcean](https://digitalocean.com/) Droplet.
+
+* faas-cli
+
+faas-cli is a command line tool used to interact with OpenFaaS, from deploying your functions to viewing logs.
+Follow the installation instructions [here](https://docs.openfaas.com/cli/install/)
+
+> You can follow along with the code used in this tutorial from this [repo](https://github.com/utsavanand2/faasd-hello)
+
+## Create a Droplet with faasd
```

Comment:
Please skip this section and point to https://docs.openfaas.com/deployment to avoid duplication (which has to be maintained) and to give the users the option to pick whatever they want.